src/java/org/apache/cassandra/service/ReplicaFilteringProtection.java

```diff
@@ -239,7 +253,21 @@ public Row onMergedRows(Row merged, Row[] versions)
                 {
                     // cache the row versions to be able to regenerate the original row iterator
                     for (int i = 0; i < versions.length; i++)
+                    {
                         builders[i].addRow(versions[i]);
+                        rowsCached++;
```

Comment:
As you can see, we only count rows toward the limit here, and not range tombstones as well. I'm not sure if this makes a difference.

Reply:
Are these limited by `tombstone_failure_threshold`?

Reply:
@adelapena I think that's at the replica? Even so, I think you're right that it provides some protection.